lib/update.py

```diff
@@ -47,12 +53,23 @@ def update_inkstitch_document(svg, selection=None):
                 update_legacy_params(EmbroideryElement(element), file_version, INKSTITCH_SVG_VERSION)
         else:
             # this is the automatic update when a legacy inkstitch svg version was recognized
-            for element in document.iterdescendants():
-                if element.tag in EMBROIDERABLE_TAGS:
-                    update_legacy_params(EmbroideryElement(element), file_version, INKSTITCH_SVG_VERSION)
+            automatic_version_update(document, file_version, INKSTITCH_SVG_VERSION)
+
         _update_inkstitch_svg_version(svg)
 
 
+def automatic_version_update(document, file_version, INKSTITCH_SVG_VERSION):
+    # make sure the user really wants to update
+    if file_version == 0:
+        do_update = RequestUpdate()
+        if do_update.cancelled is True:
+            return
```

Comment:
Maybe we should go ahead and set the version to the latest, since the user has told us not to update, like in the case of pasted objects?  Otherwise we're gonna keep asking them, which would be annoying.